### ISSUE TITLE
Relative path representations in query strings break blitz storage

### ISSUE THREAD

**Issue by paulboger** [putyourlightson/craft-blitz#343]
Relative path representations in query strings break blitz storage and can overwrite cached content.

- Plugin version: 3.10.2
- Craft version: 3.7.11

**Comment by bencroker** [comment]
Thanks for reporting this! Since this is security related, I've remove your issue description – next time, please consider contacting us directly via email (info@putyourlightson.com) with the details.

I've addressed the issue in the code. It would be extremely helpful if you could test the fix on your end.

**Comment by paulboger** [comment]
Thanks, @bencroker! Noted, and will do. We'll report back with our findings.

**Comment by bencroker** [comment]
Please do!

**Comment by paulboger** [comment]
@bencroker sorry for the delay. Following the instructions we posted last night, the same sequence only results in two cached files:
web/cache/blitz/{domain}/index.html
web/cache/blitz/{domain}/search/q=foo/index.html

So, it appears to be working all as intended. This should also resolve the related bloated cache issue which we've observed in similar environments. Let me know if we can help to any further extent. Thanks!
